src/core/tests/type_prop/scaled_dot_product_attention.cpp

```diff
@@ -12,156 +12,408 @@
+    const auto query = std::make_shared<opset13::Parameter>(element::dynamic, PartialShape{2, 3, 4});
+    const auto key = std::make_shared<opset13::Parameter>(element::dynamic, PartialShape{2, 5, 4});
+    const auto value = std::make_shared<opset13::Parameter>(element::dynamic, PartialShape{2, 5, 6});
+    const auto attention_mask = std::make_shared<opset13::Parameter>(element::dynamic, PartialShape{1, 3, 5});
+    const auto scale = std::make_shared<opset13::Parameter>(element::f32, PartialShape{});
+    auto causal = false;
+
+    const auto op =
+        std::make_shared<opset13::ScaledDotProductAttention>(query, key, value, attention_mask, scale, causal);
+    EXPECT_EQ(op->get_output_element_type(0), element::f32);
+    EXPECT_EQ(op->get_output_partial_shape(0), (PartialShape{2, 3, 6}));
+}
+
+TEST(type_prop, scaled_dot_product_attention_type_infer_4_inputs) {
+    const auto query = std::make_shared<opset13::Parameter>(element::dynamic, PartialShape{2, 3, 4});
+    const auto key = std::make_shared<opset13::Parameter>(element::f64, PartialShape{2, 5, 4});
+    const auto value = std::make_shared<opset13::Parameter>(element::dynamic, PartialShape{2, 5, 6});
+    const auto attention_mask = std::make_shared<opset13::Parameter>(element::f64, PartialShape{3, 5});
+    auto causal = false;
+
+    const auto op = std::make_shared<opset13::ScaledDotProductAttention>(query, key, value, attention_mask, causal);
+    EXPECT_EQ(op->get_output_element_type(0), element::f64);
+    EXPECT_EQ(op->get_output_partial_shape(0), (PartialShape{2, 3, 6}));
+}
+
+TEST(type_prop, scaled_dot_product_attention_type_infer_4_inputs_bool_attention) {
+    const auto query = std::make_shared<opset13::Parameter>(element::dynamic, PartialShape{2, 3, 4});
+    const auto key = std::make_shared<opset13::Parameter>(element::f64, PartialShape{2, 5, 4});
+    const auto value = std::make_shared<opset13::Parameter>(element::dynamic, PartialShape{2, 5, 6});
+    const auto attention_mask = std::make_shared<opset13::Parameter>(element::boolean, PartialShape{1, 3, 5});
     auto causal = false;
 
-    const auto gn =
+    const auto op = std::make_shared<opset13::ScaledDotProductAttention>(query, key, value, attention_mask, causal);
+    EXPECT_EQ(op->get_output_element_type(0), element::f64);
+    EXPECT_EQ(op->get_output_partial_shape(0), (PartialShape{2, 3, 6}));
+}
+
+TEST(type_prop, scaled_dot_product_attention_static_5_inputs_broadcast) {
+    const auto query = std::make_shared<opset13::Parameter>(element::f32, PartialShape{2, 1, 3, 1});
+    const auto key = std::make_shared<opset13::Parameter>(element::f32, PartialShape{1, 7, 1, 4});
+    const auto value = std::make_shared<opset13::Parameter>(element::f32, PartialShape{2, 7, 5, 1});
+    const auto attention_mask = std::make_shared<opset13::Parameter>(element::f32, PartialShape{1, 1, 1, 5});
+    const auto scale = std::make_shared<opset13::Parameter>(element::f32, PartialShape{1});
```

Comment:
SHould we be testing those if they do not match the specifications of the op?

Reply:
It seem that batch broadcasts are required by models according to https://github.com/openvinotoolkit/openvino/pull/21217#pullrequestreview-1747252695. There were also 1 additional broadcast for L shape in CPU test for L dim, however, with it disabled, models still seem to pass, so I've modified CPU test to remove it